src/node_messages.cc

```diff
@@ -0,0 +1,65 @@
+
+#include "node_messages.h"
+#include "node.h"
+#include "v8.h"
+#include "util.h"
+#include "env.h"
+#include "env-inl.h"
+
+#define TYPE_ERROR(msg) env->ThrowTypeError(msg)
+
+// Used to be a macro, hence the uppercase name.
+#define DEFINE_MSG_CONSTANT(target, constant)                                 \
+  do {                                                                        \
+    v8::Isolate* isolate = target->GetIsolate();                              \
+    v8::Local<v8::String> constant_name =                                     \
+        v8::String::NewFromUtf8(isolate, #constant);                          \
+    v8::Local<v8::Number> constant_value =                                    \
+        v8::Number::New(isolate, static_cast<double>(MSG_ ## constant));      \
+    v8::PropertyAttribute constant_attributes =                               \
+        static_cast<v8::PropertyAttribute>(v8::ReadOnly | v8::DontDelete);    \
+    (target)->ForceSet(constant_name, constant_value, constant_attributes);   \
+  }                                                                           \
+  while (0)
+
+namespace node {
+
+using v8::Local;
+using v8::Object;
+using v8::String;
+using v8::Value;
+using v8::FunctionCallbackInfo;
+
+static void NodeMsg(const FunctionCallbackInfo<Value>& args) {
```

Comment:
Given that these are exceptional cases, I'm not sure the additional cost is going to matter much. However, once I'm a bit further along, I want to explore caching for the most common error cases so that the most commonly used messages only have to be pulled from the runtime once